chainer/functions/array/copy.py

```diff
@@ -36,24 +41,38 @@ def copy(x, dst):
     Args:
         x (:class:`~chainer.Variable` or :ref:`ndarray`):
             Variable to be copied.
-        dst (int): Target device specifier.
+        dst: Target device specifier.
 
     Returns:
         ~chainer.Variable: Output variable.
 
     .. admonition:: Example
 
         >>> import chainer.backends.cuda as cuda
-        >>> x = np.random.uniform(-1, 1, (5, 10))
-        >>> cuda.get_device_from_array(x).id
-        -1
-        >>> y = F.copy(x, 0) # from host to device0
-        >>> cuda.get_device_from_array(y.array).id
-        0
-        >>> z = F.copy(y, -1) # from device0 to host
-        >>> cuda.get_device_from_array(z.array).id
-        -1
+        >>> x_arr = np.random.uniform(-1, 1, (5, 10))
+        >>> x = chainer.Variable(x_arr)
+        >>> x.device
+        '@numpy'
+        >>> y = F.copy(x, '@cupy:0') # from CPU (NumPy) to GPU 0 (CuPy)
+        >>> y.device
+        '@cupy:0'
+
+    .. note::
+        Copies between non-ChainerX devices and ChainerX devices are not
+        supported.
 
     """
-    y, = Copy(dst).apply((x,))
+    in_device = backend.get_device_from_array(
+        x.array if isinstance(x, chainer.Variable) else x)
+    out_device = chainer.get_device(dst)
+
+    is_chainerx = in_device.xp is chainerx
+    if is_chainerx != (out_device.xp is chainerx):
+        raise RuntimeError(
+            'F.copy does not support a copy between a non-ChainerX device and '
```

Comment:
A major nitpick but I think the following is a more common way to put it.
```suggestion
            'F.copy does not support copies between non-ChainerX devices and '
```

Reply:
Thanks, fixed.